src/core/server/logging/layouts/pattern_layout.test.ts

```diff
@@ -118,6 +119,45 @@ test('`format()` correctly formats record with custom pattern.', () => {
   }
 });
 
+test('`format()` correctly formats record with meta data.', () => {
+  const layout = new PatternLayout();
+
+  expect(
+    layout.format({
+      context: 'context-meta',
+      level: LogLevel.Debug,
+      message: 'message-meta',
+      timestamp,
+      pid: 5355,
+      meta: {
+        from: 'v7',
+        to: 'v8',
+      },
+    })
+  ).toBe('[2012-02-01T14:30:22.011Z][DEBUG][context-meta][{"from":"v7","to":"v8"}] message-meta');
+
+  expect(
+    layout.format({
+      context: 'context-meta',
+      level: LogLevel.Debug,
+      message: 'message-meta',
+      timestamp,
+      pid: 5355,
+      meta: {},
+    })
+  ).toBe('[2012-02-01T14:30:22.011Z][DEBUG][context-meta][{}] message-meta');
```

Comment:
Is it useful to include meta in the layout if it's an empty object?

Reply:
an object could be empty or stringified to an empty object. In this case, I'd rather show it explicitly in logs. Otherwise, we will have complaints on
meta-object disappearing from the logs.
```js
let something = undefined;
// ....
log('message', { something }); // absence from the log since stringified to an empty object
```